enterprise/templates/enterprise/grant_data_sharing_permissions.html

```diff
@@ -31,7 +31,11 @@
                         </h2>
                     </div>
                     <span class="text">
-                        ${_("In order to link your account, we need consent to share your course data with {provider}.").format(provider=sso_provider)}
+                        % if not course_id:
```

Comment:
I think it would be better to have a specific parameter to control this behavior instead of using `course_id` as the switch